src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -1040,6 +1046,13 @@ export default class DefaultAudioVideoController
   }
 
   private async actionUpdateWithRenegotiation(notify: boolean): Promise<void> {
+    // `actionUpdateWithRenegotiation` contains a significant portion of asynchronous tasks, so we need
+    // to explicitly defer any task operation which may be performed on the event queue that may modify
+    // mutable state needed to be consistent over the course of the update. The operations above do not
+    // need this protection because they are synchronous.
+    this.monitorTask.pauseResubscribeCheck();
+    this.receiveIndexTask.pauseIngestion();
+
```

Comment:
race condition can still happen in other task like `ReceiveVideoInputTask` right?

Reply:
With `integrateUplinkPolicyDecision` ? I'm not too concerned about that impact since we are mostly relying on avg bitrate anyways.

Reply:
No, it can cause video to fail when calling choose video input multiple when the video is on (https://github.com/aws/amazon-chime-sdk-js/blob/master/src/task/ReceiveVideoInputTask.ts#L32). See this issue https://github.com/aws/amazon-chime-sdk-js/issues/1609